### PR TITLE
Update the pause image to build for arm64 and ppc64le

### LINKED ISSUE #22683: Creating pod infra container for non-intel archs
Hi,

In order to run kubernertes on non-intel arch, running pod infra container is a big issue. As the official container image i.e. gcr.io/google_containers/pause is available for intel only (Correct me if I am wrong). I have been able to create the image for ppc64le arch too, but not sure how to get that into official repos. Since, there is no namespace attached to the image, how should we identify the image based on arch?

### PR DESCRIPTION
**Pull request by luxas**
Fixes #22683

### CODE REVIEW

File: build/pause/Makefile [hunk 1]
```diff
@@ -14,18 +14,45 @@
...
+	docker run -it -v $(TEMP_DIR):/build golang:$(GOLANG_VERSION) /bin/bash -c \
+		"cd /build && CGO_ENABLED=0 GOARM=6 GOARCH=$(ARCH) go build -a -installsuffix cgo -ldflags '-w' ./pause.go"
+
+ifeq ($(ARCH),amd64)
+# Run goupx for amd64 and arm
+	docker run -it -v $(TEMP_DIR):/build golang:$(GOLANG_VERSION) /bin/bash -c \
+		"cd /build && apt-get update && apt-get install -y upx \
+		&& go get github.com/pwaller/goupx && goupx pause"
+endif
+ifeq ($(ARCH),arm)
```
Comment (luxas) [inline build/pause/Makefile]: I hate this duplication. @zmerlynn Do you know an `if [[ cond1 || cond2 ]]; then` clause in `make`?

Comment (zmerlynn) [inline build/pause/Makefile]: No: http://stackoverflow.com/questions/7656425/makefile-ifeq-logical-or

**Comment by Pensu** [comment]
@luxas Shouldn't it be make push ARCH=arm64 and make push ARCH=**ppc64le**!

**Comment by luxas** [comment]
@Pensu Oops, copy-paste error :)
Updated the comment

**Comment by ihmccreery** [comment]
@zmerlynn Can you please take a look at this, since you've got context on #22686?

**Comment by zmerlynn** [comment]
@luxas: Pushed.

**Comment by luxas** [comment]
Many thanks @zmerlynn :smile:!

@Pensu Feel free to try it out on your `ppc64le` machines!

**Comment by Pensu** [comment]
@luxas Sorry for the delayed reply, I verified it but forgot to comment. It's working completely fine. Thanks!

**Comment by luxas** [comment]
@Pensu Please follow #23931 closely. When that one is in, one may run http://kubernetes.io/docs/getting-started-guides/docker/ and it should just work. Only `s/amd64/ppc64le/` in the image name and you're done.

**Pull request merged**